src/Extension/FrontMatter/FrontMatterParser.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the league/commonmark package.
+ *
+ * (c) Colin O'Dell <colinodell@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace League\CommonMark\Extension\FrontMatter;
+
+use League\CommonMark\Event\DocumentPreParsedEvent;
+use League\CommonMark\Extension\FrontMatter\Yaml\FrontMatterParserInterface;
+use League\CommonMark\Input\MarkdownInput;
+use League\CommonMark\Parser\Cursor;
+
+final class FrontMatterParser
+{
+    /**
+     * @var FrontMatterParserInterface
+     *
+     * @psalm-readonly
+     */
+    private $yamlParser;
+
+    private const REGEX_FRONT_MATTER = '/^---\\n.*\\n---\n/s';
```

Comment:
From my work on https://www.drupal.org/project/drupal/issues/3064854, the following regex is probably a little more forgiving in how newlines can be interpreted across OS:

```php
/**
 * The separator used to indicate front matter data.
 *
 * @var string
 */
 const SEPARATOR = '---';

/**
 * The regular expression used to extract the YAML front matter content.
 *
 * @var string
 */
const REGEXP = '/\A(' . self::SEPARATOR . '(.*?)?\R' . self::SEPARATOR . ')(\R.*)?\Z/s';
```